pkg/kubectl/cmd/create.go

```diff
@@ -360,5 +363,43 @@ func RunCreateSubcommand(f cmdutil.Factory, cmd *cobra.Command, out io.Writer, o
 		return nil
 	}
 
-	return f.PrintObject(cmd, false, mapper, obj, out)
+	return ensurePrintVersionedObject(f, mapper, cmd, obj, out)
+}
+
+// ensurePrintVersionedObject receives a runtime.Object and ensures that it has been converted to an external version
+// before passing it through to the printer stack.
+func ensurePrintVersionedObject(f cmdutil.Factory, mapper meta.RESTMapper, cmd *cobra.Command, obj kruntime.Object, out io.Writer) error {
+	options := cmdutil.ExtractCmdPrintOptions(cmd, false)
+	printer, err := f.PrinterForOptions(options)
+	if err != nil {
+		return err
+	}
+
+	// we don't care about converting the object if not printing in a generic
+	// format or if the object we are dealing with is unstructured.
+	_, isUnstructured := obj.(*unstructured.Unstructured)
+	if isUnstructured || !printer.IsGeneric() {
+		return f.PrintObject(cmd, false, obj, out)
+	}
+
+	gvks, _, err := legacyscheme.Scheme.ObjectKinds(obj)
+	if err != nil {
+		return err
+	}
+
+	preferredVersion := []string{}
+	for _, gvk := range gvks {
+		if len(gvk.Version) > 0 && gvk.Version != kruntime.APIVersionInternal {
+			preferredVersion = append(preferredVersion, gvk.Version)
+			break
+		}
+	}
+
+	mapping, err := mapper.RESTMapping(gvks[0].GroupKind(), preferredVersion...)
```

Comment:
this looks very wrong.  The object converter should be `legacyscheme.Scheme`.  The typer (what you used to get the groupversion kinds), should also be `legacyscheme.Scheme`